helix-core/src/main/java/org/apache/helix/HelixCloudProperty.java

```diff
@@ -0,0 +1,179 @@
+          try {
+            InputStream stream = Thread.currentThread().getContextClassLoader().getResourceAsStream(AZURE_CLOUD_PROPERTY_FILE);
+            azureProperties.load(stream);
+          } catch (Exception e) {
+            String errMsg = "failed to open Helix Azure cloud properties file: " + AZURE_CLOUD_PROPERTY_FILE;
+            throw new IllegalArgumentException(errMsg, e);
+          }
+          LOG.info("Successfully loaded Helix Azure cloud properties: " + azureProperties);
+          setCloudInfoSources(Collections.singletonList(azureProperties.getProperty(CLOUD_INFO_SOURCE)));
+          setCloudInfoProcessorName(azureProperties.getProperty(CLOUD_INFO_PROCESSFOR_NAME));
+          setCloudMaxRetry(azureProperties.getProperty(CLOUD_MAX_RETRY));
+          setCloudConnectionTimeout(azureProperties.getProperty(CONNECTION_TIMEOUT_MS));
+          setCloudRequestTimeout(azureProperties.getProperty(REQUEST_TIMEOUT_MS));
+          break;
+        case CUSTOMIZED:
+          setCloudInfoSources(cloudConfig.getCloudInfoSources());
+          setCloudInfoProcessorName(cloudConfig.getCloudInfoProcessorName());
+          break;
+        default:
+          LOG.info("Unsupported cloud provider: " + cloudConfig.getCloudProvider());
+          throw new HelixException(String.format("Unsupported cloud provider: %s", cloudConfig.getCloudProvider()));
+      }
+    }
+  }
+
+  public boolean getCloudEnabled() {
+    return _isCloudEnabled;
+  }
+
+  public String getCloudId() {
+    return _cloudId;
+  }
+
+  public String getCloudProvider() {
+    return _cloudProvider;
+  }
+
+  public List<String> getCloudInfoSources() {
+    return _cloudInfoSources;
+  }
+
+  public String getCloudInfoProcessorName() {
+    return _cloudInfoProcessorName;
+  }
+
+  public int getCloudMaxRetry() {
+    return _cloudMaxRetry;
+  }
+
+  public long getCloudConnectionTimeout() {
+    return _cloudConnectionTimeout;
+  }
+
+  public long getCloudRequestTimeout() {
+    return _cloudRequestTimeout;
+  }
+
+  public Properties getCustomizedCloudProperties() {
+    return _customizedCloudProperties;
+  }
+
+  public void setCloudEndabled(boolean isCloudEnabled) {
+    _isCloudEnabled = isCloudEnabled;
+  }
+
+  public void setCloudId(String cloudId) {
+    _cloudId = cloudId;
+  }
+
+  public void setCloudProvider(String cloudProvider) {
+    _cloudProvider = cloudProvider;
+  }
+
+  public void setCloudInfoSources(List<String> sources) {
+    _cloudInfoSources = sources;
+  }
+
+  public void setCloudInfoProcessorName(String cloudInfoProcessorName) {
+    _cloudInfoProcessorName = cloudInfoProcessorName;
+  }
+
+  public void setCloudMaxRetry(String cloudMaxRetry) {
+    _cloudMaxRetry = Integer.valueOf(cloudMaxRetry);
+  }
+
+  public void setCloudConnectionTimeout(String cloudConnectionTimeout) {
+    _cloudConnectionTimeout = Long.valueOf(cloudConnectionTimeout);
+  }
+
+  public void setCloudRequestTimeout(String cloudRequestTimeout) {
```

Comment:
From the comment of the class: the fields could be changed by clients. So I think the parameter type could be `long` so it is consistent with the field type. And clients don't have to convert a long type timeout to a string.